S0 add heroku deployment

dhilton opened the pull request:
__What__

This pull request creates the initial infrastructure for demoing the EQ(electronic questionnaire) project architecture. It consists of a terraform definition for two Heroku apps and a database attached to one. This is the first iteration of what we are trying to achieve with EQ infrastructure.

__How to test this PR__

On a linux or osx system, clone the repo and check out this branch. Following the instructions in the README, check that you can install Terraform and replicate the deployment of two heroku applications and verify that the database is attached to the author application. You may also wish to check that the correct environment variables have been injected into their respective environments.

__Who can test__

Anyone on the EQ team apart from @dhilton

dhilton commented on README.md:
```diff
@@ -1 +1,27 @@
+2. Ensure that you have signed up for a heroku account and have generated an API
+key.
+
+3. Copy `terraform.tfvars.example` to `terraform.tfvars` and replace your heroku
```
I don't think we need the Heroku api key do we?

warrenbailey replied:
Good point, copy and paste issue there

dhilton commented on terraform.tfvars.example:
```diff
@@ -0,0 +1,3 @@
+aws_access_key="XXXXXXXXXXXX"
+aws_secret_key="XXXXXXXXXXXX"
+aws_key_pair="digital-eq-keypair"
```
Is this key pair used by this deployment? The referenced keypair above is `pre-prod.pem` ?

warrenbailey replied:
Yep you're right, should be pre-prod.pem. I'll change it

collisdigital commented:
Testing Slack integration again...

dhilton commented:
@warrenbailey I've refactored the deployment code a little, making the bash parts a bit more succinct and removed the need for the local template dance. 

I've pushed a new branch to the ansible eq-messaging repo to fix the variables issue - the last part is the hosts file not being correctly set in this terraform run; the bash fails on multilines but the last commit should address that.

Amended branch: https://github.com/ONSdigital/eq-messaging/tree/EQ-18-Amend-variables

warrenbailey commented:
@dhilton that's looks a lot better. I didn't like that local file stuff, this is much better :+1:

warrenbailey commented:
Tested and working.

dhilton commented:
LGTM :+1:
